src/importexport/musicxml/internal/musicxml/importmxmlpass2.cpp

```diff
@@ -2328,6 +2328,13 @@ void MusicXMLParserPass2::staffDetails(const QString& partId)
         t->setFrets(25);      // sensible default
     }
 
+    QString visible = _e.attributes().value("print-object").toString();
+    if (visible == "no") {
+        _score->staff(staffIdx)->setVisible(false);
+    } else if (visible != "yes") {
+        _logger->logError(QString("print-object should be \"yes\" or \"no\""));
+    }
```

Comment:
If possible without major restructuring, once the import is nearly complete and all staves have been made visible or invisible in MuseScore's internal structures, you should check whether all staves in each `Part` are invisible and if so make the `Part` itself invisible, and make all the staves inside it visible again (they won't be shown if the `Part` is invisible. This will help to ensure that the import matches the export.

Reply:
I think a better fix would be to make the `Part` invisible when all its staves are invisible. This is the current behaviour:
![image](https://user-images.githubusercontent.com/88136054/174942769-0bc5687c-a99d-42c1-a879-c94f28f39058.png)

Do you think there would be edge cases where this would not be desired? (@Tantacrul too)

Reply:
I think you're right. We'd just need to make sure we avoid janky situations.
 

- If all your staves are empty and you add a new stave, it should be **visible** by default, thereby turning back on the visibility of your part
- (visa-versa) If you remove that visible stave, which means there are no more visible staves, then the part turns back off
- Turning on/off the part visibility affects the visibility of **all** its staves in all circumstances (to avoid loads of additional unnecessary logic)

Reply:
But it isn't necessary to do any of the above unless there is a good technical reason for it. From a UX point of view, it is fine as it is. However, changing it t be the way @HemantAntony suggest's isn't a problem either.

Reply:
It's probably best to do as @HemantAntony suggests and make the `Part` invisible if all of its staves are invisible, and the staves invisible if the Part is made invisible. There are other ways it could work but that way is probably the least complicated option.

Reply:
I sadly had to abandon this way of doing it though this is the ideal way of doing it. It turns out doing this would require some larger changes to the existing code. So I'm doing it the way @shoogle proposed. So the Part won't be hidden if all its staves are